RecoLocalCalo/HGCalRecProducers/plugins/alpaka/HGCalLayerClustersSoAAlgoWrapper.dev.cc

```diff
@@ -0,0 +1,227 @@
+#include "RecoLocalCalo/HGCalRecProducers/interface/HGCalTilesConstants.h"
+
+#include "HGCalLayerClustersSoAAlgoWrapper.h"
+
+#include "CLUEAlgoAlpaka.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+
+  // Set energy and number of hits in each clusters
+  class HGCalLayerClustersSoAAlgoKernelEnergy {
+  public:
+    template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc,
+                                  const unsigned int numer_of_clusters,
+                                  const HGCalSoARecHitsDeviceCollection::ConstView input_rechits_soa,
+                                  const HGCalSoARecHitsExtraDeviceCollection::ConstView input_clusters_soa,
+                                  HGCalSoAClustersDeviceCollection::View outputs) const {
+      // make a strided loop over the kernel grid, covering up to "size" elements
+      for (int32_t i : uniform_elements(acc, input_rechits_soa.metadata().size())) {
+        // Skip unassigned rechits
+        if (input_clusters_soa[i].clusterIndex() == -1) {
```

Comment:
Is there a constant that can be used instead of `-1` to identify the invalid clusters ?

Reply:
0 is a valid `clusterIndex`. Since the type of this variable is `int`, I guess -1 is a reasonable invalid value.

Reply:
then maybe
```
constexpr const int kInvalidCluster = -1;
```
?